platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/FilterInjector.java

```diff
@@ -100,12 +107,17 @@ private void checkForMissedServletContexts() {
       }
       Collection<ServiceReference<ServletContext>> references =
           context.getServiceReferences(ServletContext.class, null);
+
+      List<Filter> allFilterList = new ArrayList<>();
```

Comment:
✏️  Avoid creating a temporary list just to iterate over it later.
Maybe replace the loop later with: 
```suggestion
      Stream.of(filterList, referenceFilterList)
        .flatMap(List::stream)
        .map(Object::getClass)
        .map(Class::toString)
        .map(service::getFilterRegistration)
        .filter(Objects::nonNull)
        .forEach(r -> {
          LOGGER.error(
              "Security filter failed to start in time to inject itself into {} {}. This means the {} servlet will not properly attach the user subject to requests. A system restart is recommended.",
              refBundle.getSymbolicName(),
              refBundle.getBundleId(),
              refBundle.getSymbolicName());
        });
```

Reply:
Yep, I know you can do it with streams, but just creating the temporary list and using that in a traditional loop is around 30% faster than this method and that is important for code that gets hit on every single request.

Reply:
Then why not pre-combine these 2 lists into one in the constructor? If this is to be called often, we are creating temporary collections to be garbage collected each time ...

Reply:
In addition you could think of actually creating that list with the class name of each filter to speed up even more that loop and avoid retrieving the class and its name each time

Reply:
If your goal is to keep the list dynamic, then consider creating a combined iterator. Apache and Guava has utility classes that does that

Reply:
❔  I don't think this code is called on every single request. Should be called to inject filter around bundle startup I'd think and actual requests just see the filters.

Reply:
We can't precombine the lists because blueprint keeps the reference one up to date. This particular code is not called on every request, but the one below is. The comment was left on both.

Reply:
You could still keep them as is and use a combined iterator thus avoiding the creation of a temporary list